src/tests/integration/test_hevm_prove.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+import sys
+from distutils.dir_util import copy_tree
+from typing import TYPE_CHECKING
+
+import pytest
+from filelock import FileLock
+from pyk.kore.rpc import kore_server
+from pyk.proof import APRProof
+from pyk.proof.reachability import APRFailureInfo
+from pyk.utils import run_process, single
+
+from kontrol.foundry import Foundry
+from kontrol.kompile import foundry_kompile
+from kontrol.options import ProveOptions, RPCOptions
+from kontrol.prove import foundry_prove
+
+from .utils import TEST_DATA_DIR
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from pathlib import Path
+    from typing import Final
+
+    from pyk.kore.rpc import KoreServer
+    from pyk.proof.proof import Proof
+    from pyk.utils import BugReport
+    from pytest import TempPathFactory
+
+
+FORGE_STD_REF: Final = '75f1746'
+
+
+sys.setrecursionlimit(10**7)
+
+
+@pytest.fixture(scope='module')
+def server(foundry: Foundry, no_use_booster: bool) -> Iterator[KoreServer]:
+    llvm_definition_dir = foundry.out / 'kompiled' / 'llvm-library' if not no_use_booster else None
+    kore_rpc_command = ('kore-rpc-booster',) if not no_use_booster else ('kore-rpc',)
+
+    yield kore_server(
+        definition_dir=foundry.kevm.definition_dir,
+        llvm_definition_dir=llvm_definition_dir,
+        module_name=foundry.kevm.main_module,
+        command=kore_rpc_command,
+        smt_timeout=300,
+        smt_retry_limit=10,
+        fallback_on=None,
+        interim_simplification=None,
+        no_post_exec_simplify=None,
+    )
+
+
+@pytest.fixture(scope='session')
+def foundry(foundry_root_dir: Path | None, tmp_path_factory: TempPathFactory, worker_id: str) -> Foundry:
+    if foundry_root_dir:
+        return Foundry(foundry_root_dir)
+
+    if worker_id == 'master':
+        root_tmp_dir = tmp_path_factory.getbasetemp()
+    else:
+        root_tmp_dir = tmp_path_factory.getbasetemp().parent
+
+    foundry_root = root_tmp_dir / 'foundry'
+    with FileLock(str(foundry_root) + '.lock'):
+        if not foundry_root.is_dir():
+            copy_tree(str(TEST_DATA_DIR / 'foundry'), str(foundry_root))
+
+            run_process(['forge', 'install', '--no-git', f'foundry-rs/forge-std@{FORGE_STD_REF}'], cwd=foundry_root)
+            run_process(['forge', 'build'], cwd=foundry_root)
+
+            foundry_kompile(
+                foundry=Foundry(foundry_root),
+                includes=(),
+                requires=[str(TEST_DATA_DIR / 'lemmas.k')],
+                imports=['LoopsTest:SUM-TO-N-INVARIANT'],
```

Comment:
Since `LoopsTest` is not being run here, we probably don't need to import it. I wonder if we're rekompiling files for hevm tests, or are they being shared with Foundry tests?